A modest proposal to install the bigtable client (and Abseil too!)

coryan opened the pull request:
This PR is intended for discussion, it is likely that we will go back and do something different.  The relevant bug is #24 .  In this PR we install the bigtable client libraries and compile a trivial example against the installed libraries.

The problem was: how to install Abseil?  We wrap the Abseil CMake files in our own `CMakeLists.txt` file and modify their functions to include the install targets.

coryan commented:
Any thoughts?

mbrukman commented on bigtable/tests/Makefile:
```diff
@@ -0,0 +1,26 @@
+# nor is it a general solution on how to build Makefiles for google-cloud-cpp.
+# It is intended to test the installation, many things are hard-coded in this
+# test (e.g., PREFIX, the "d" suffix for libraries, the compiler options), which
+# would be configured by any
```
Unfinished sentence?

coryan replied:
Fixed.

mbrukman commented:
Also, this PR has conflicting files.

coryan commented on .gitmodules:
```diff
@@ -10,7 +10,7 @@
 	ignore = dirty
 [submodule "third_party/abseil"]
-	path = third_party/abseil
+	path = third_party/abseil/abseil-cpp
```
They will need to `git submodule update --init` again.

coryan commented on bigtable/tests/Makefile:
```diff
@@ -0,0 +1,26 @@
+# A simple makefile to test the `install` target.
```
Because neither #45 nor #44 are done yet.  Writing a simple Makefile that is hard-coded is easier than writing a CMakeLists.txt that is kind of hard-coded.  Happy to change this if we agree on the rest of the proposal though.

coryan replied:
Added a comment to declare what we should do here.

coryan commented on .travis.yml:
```diff
@@ -48,9 +48,9 @@ matrix:
-      # using a pre-installed gRPC package.  To do so, it builds the normal
-      # Docker image with all the build tools, and uses that image to build and
+    - # The following entries two are fairly specialized.  They build the clients
+      # using a pre-installed gRPC package.  To do so, they builds the normal
```
Fixed?  Actually completely changed during the rebase.

coryan commented:
Rebased to resolve the conflicts.  Also addressed most of your (@mbrukman) comments.  I think we should discuss in person, because all the options in these area are bad.

coryan commented:
I think #225 is a better approach. Closing.
